### DISCUSSION TITLE
Maximum filesize (comparing text files)

### DISCUSSION THREAD

**Discussion by yamhillbilly** [WinMerge/winmerge#2019] [Q&A]
Can people comment on max filesize (comparing 2 text files) that WinMerge can handle?  I'd be looking at comparing two text files that might be something like 7 million lines of text, 800 MB on a Windows system.  I imagine the answer depends on factors beyond the WinMerge software, but I wanted to at least check whether files of this size are likely workable.  Thanks!

**Comment by sdottaka** [comment]
First, the 32-bit version of WinMerge has a memory limit of 2GB.
WinMerge stores file contents in memory in UTF-16, which consumes twice as much memory as the file size. Additionally, diff calculations require memory usage proportional to the number of lines of text. Therefore, you will not be able to open two files of 800MB in size with the 32-bit version of WinMerge.
The 64-bit version of WinMerge does not have a 2GB limit and can theoretically handle 2 billion lines. If you have enough free memory, you should be able to compare 800MB files. However, diff calculations are very time-consuming, so you may have to wait for a very long time.

**Reply by yamhillbilly** [reply]
Thank you @sdottaka for that detailed and very helpful response.
